### PR TITLE
Feature/cmrarc 595 server side functionality for paging tool.

### PR DESCRIPTION
**Pull request by htranho**

### CODE REVIEW

File: app/controllers/records_controller.rb [hunk 1]
```diff
@@ -5,16 +5,67 @@ class RecordsController < ApplicationController
   before_action :ensure_curation
   before_action :admin_only, only: [:stop_updates, :allow_updates, :revert]
   before_action :find_record, only: [:show, :complete, :update, :stop_updates, :allow_updates, :revert, :copy_prior_recommendations]
-  before_action :filtered_records, only: :finished
+  before_action :filtered_records, only: [:finished]
```
Comment (htranho) [inline app/controllers/records_controller.rb:8]: Remove bracket

File: Gemfile [hunk 2]
```diff
@@ -94,4 +94,5 @@ group :development do
   gem 'rdoc'
   gem 'rubyzip', '1.3.0'
   gem 'rails-erd'
+  gem 'kaminari'
```
Comment (cgokey) [inline Gemfile:97]: Remove this.

File: app/controllers/records_controller.rb [hunk 3]
```diff
@@ -1,20 +1,176 @@
...
   include RecordHelper
+  include SiteHelper
 
   before_action :authenticate_user!
   before_action :ensure_curation
   before_action :admin_only, only: [:stop_updates, :allow_updates, :revert]
   before_action :find_record, only: [:show, :complete, :update, :stop_updates, :allow_updates, :revert, :copy_prior_recommendations]
-  before_action :filtered_records, only: :finished
 
+  def find_records_json_active_record
```
Comment (cgokey) [inline app/controllers/records_controller.rb:10]: remove this.

File: app/helpers/site_helper.rb [hunk 4]
```diff
@@ -20,4 +21,24 @@ def title_for_in_review_records()
       'In ARC Review Records'
     end
   end
+
```
Comment (cgokey) [inline app/helpers/site_helper.rb:24]: remove

File: app/javascript/components/data/query_response.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
+const result =
```
Comment (cgokey) [inline app/javascript/components/data/query_response.js:1]: remove

File: app/javascript/components/images/nasa-logo_hover.png [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
```
Comment (cgokey) [inline app/javascript/components/images/nasa-logo_hover.png:1]: re-download this

File: app/javascript/components/images/nasa-logo_hover_2x.png [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
```
Comment (cgokey) [inline app/javascript/components/images/nasa-logo_hover_2x.png:1]: re-download

File: app/javascript/components/index.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
```
Comment (cgokey) [inline app/javascript/components/index.js:1]: remove

File: app/javascript/components/view/HeaderView.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+      render() {
+        return (
+          <thead>
+          <tr>
+            {this.props.headers.map(header => (
+              <th className={header.sortable ? "sortableHeader" : "header"} style={{width: header.width}} key={header.field}
+                  onClick={() => {
+                    if (header.sortable) {
+                      this.toggleTriangle(header)
```
Comment (cgokey) [inline app/javascript/components/view/HeaderView.js:37]: move this into the vm.

File: app/javascript/components/view/PaginationTableView.js [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+      return null
+    }
+
+    componentDidMount() {
+      this.viewModel.daac = this.getDaac()
+      this.viewModel.campaign = this.getCampaign()
+      this.viewModel.selectPage(this.viewModel.currentPage)
+    }
+
+    checkAll(table_name)
```
Comment (cgokey) [inline app/javascript/components/view/PaginationTableView.js:46]: rename to selectAll

File: app/javascript/components/view/PaginationTableView.js [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+            let pos = value.indexOf("T")
+            if (pos > -1) {
+              value = value.substring(0, pos)
+            }
+            column = <td key={header.field}>{value}</td>
+          }
+        }
+        if (header.field == 'selection') {
+          column = <td key={header.field} className="center_text"><input onClick={() => {
+            this.toggleButtons(this.props.formId)
```
Comment (cgokey) [inline app/javascript/components/view/PaginationTableView.js:123]: just call toggleAllButtons!

**Review by cgokey — APPROVED** [review]

**Pull request merged**